scapy/utils.py

```diff
@@ -1904,6 +1904,241 @@ def write_packet(self,
             caplen=caplen, wirelen=wirelen
         )
 
+@conf.commands.register
+def rderf(filename, count=-1):
+    # type: (Union[IO[bytes], str], int) -> PacketList
+    """Read a ERF file and return a packet list
+
+    :param count: read only <count> packets
+    """
+    with ERFEthernetReader(filename) as fdesc:
+        return fdesc.read_all(count=count)
+
+class ERFEthernetReader(_SuperSocket):
+    
+    def __init__(self, filename):  # type: ignore
+        # type: (str) -> None
+        self.filename, self.f = self.open(filename)
+
+    def _convert_erf_timestamp(self, time):
+        lts = time
+        sec = lts >> 32
+        whole_sec = EDecimal(0xffffffff)
+        frac_sec = lts & 0xffffffff
+        frac_sec *= 10**9
+        frac_sec += (frac_sec & 0x80000000) << 1
+        frac_sec >>= 32
+        power = Decimal(10) ** Decimal(-9)
+        return EDecimal(sec + power * frac_sec)
+
+    def __enter__(self):
+        # type: () -> ERFEthernetReader
+        return self
+    
+    def __exit__(self, exc_type, exc_value, tracback):
+        # type: (Optional[Any], Optional[Any], Optional[Any]) -> None
+        self.f.close()
+
+    @staticmethod
+    def open(fname  # type: Union[IO[bytes], str]
+             ):
+        # type: (...) -> Tuple[str, _ByteStream, bytes]
+        """Open (if necessary) filename"""
+        if isinstance(fname, str):
+            filename = fname
+            if filename.endswith(".gz"):
+                fdesc = gzip.open(filename, "rb")  # type: _ByteStream
+            else:
+                fdesc = open(filename, "rb")
+
+        else:
+            fdesc = fname
+            filename = getattr(fdesc, "name", "No name")
+        return filename, fdesc
+
+    def read_all(self, count=-1):
+        # type: (int) -> PacketList
+        res = self._read_all(count)
+        from scapy import plist
+        return plist.PacketList(res, name=os.path.basename(self.filename))
+
+    def _read_all(self, count=-1):
+        # type: (int) -> List[Packet]
+        """return a list of all packets in the ERF file
+        """
+        res = []  # type: List[Packet]
+        while count != 0:
+            count -= 1
+            try:
+                p = self.read_packet()  # type: Packet
+            except EOFError:
+                break
+            res.append(p)
+        return res
+
+    def read_packet(self, size=MTU):
+        # General ERF Header have exactly 16 bytes
+        hdr = self.f.read(16)
+        if len(hdr) < 16:
+            raise EOFError
+
+        # The timestamp is in little-endian byte-order
+        time = struct.unpack( '<Q', hdr[:8] )[0]
+        # The rest is in big-endian byte-order
+        type, flags, rlen, lctr, wlen = struct.unpack( '>BBHHH', hdr[8:] )
+        # Check if the type != 0x02, type Ethernet
```

Comment:
I think we should add a comment to say that we ignore `flags` and `lctr` and tell the reason (it's ERF specific and out of scapy interest)